Fix/1452

JokerJohn opened the pull request:
Add a minimal  test case of  `IncrementalFixedLagSmoother`  in `gtsam/gtsam_unstable/examples` folder using real-world data to showcase how setting `findUnusedFactorSlots = true` addresses the issue #1452 in GTSAM.

dellaert committed:
Check issue 1452

JokerJohn committed:
add IncrementalFixedLagExample to showcase how setting  addresses the issue #1452

JokerJohn committed:
add IncrementalFixedLagExample to show the issue #1452

dellaert reviewed (commented):
I already added the text file you provided in examples/Data/. Is the file you provided different?

dellaert commented on gtsam_unstable/examples/IncrementalFixedLagSmootherExample.cpp:
```diff
@@ -0,0 +1,273 @@
+ */
+int main() {
+    // Path to the test data file
+    ifstream infile("IncrementalFixedLagSmootherTestData.txt");
```
Could you load the file as in the test I provided in this branch?

JokerJohn replied:
Resolved.

dellaert commented on gtsam_unstable/examples/IncrementalFixedLagSmootherExample.cpp:
```diff
@@ -0,0 +1,273 @@
+    saveG2oGraph(smoother.getFactors(), currentEstimate, path + "isam", lineCount);
+    infile.close();
+    return 0;
+}
```
Add a new line at end of file, please

JokerJohn replied:
Resolved.

dellaert commented:
You’re merging this into the branch I created, which added this as a unit test. I don’t think we need both, so could you also delete the test in this PR? Also look at how I load the file, and perhaps how I now use `emplace_shared`.

JokerJohn committed:
Move the example to the appropriate folder, and modify the code for reading files and adding factors.

JokerJohn committed:
add a new line at the end of file

JokerJohn commented:
> You’re merging this into the branch I created, which added this as a unit test. I don’t think we need both, so could you also delete the test in this PR? Also look at how I load the file, and perhaps how I now use `emplace_shared`.

Sorry, I just noticed that you have already provided the data file `issue1452.txt` in the `Data` directory. I have moved this example to the appropriate folder and modified the corresponding file reading and factor adding code.

JokerJohn committed:
add a new line at the end of file

dellaert commented:
> You’re merging this into the branch I created, which added this as a unit test. I don’t think we need both, so could you also delete the test in this PR? Also look at how I load the file, and perhaps how I now use `emplace_shared`.

Ping on this comment :-)

JokerJohn committed:
remove example

dellaert commented:
Hi @JokerJohn - I actually liked your example - I was asking to remove the test :-) 
Feel free to re-open

JokerJohn commented:
> Hi @JokerJohn - I actually liked your example - I was asking to remove the test :-) Feel free to re-open

Sorry，I  saw the unit test code for issue 1452, which has already incorporated the example code, so I closed this PR. If there's a need to change it into an example case, I can reopen this PR.

JokerJohn committed:
add IncrementalFixedLagSmootherExample

dellaert commented:
Excellent, but could you then remove the (slow and redundant) unit test that basically does the same? Your example shows (a) it works, (b) a useful example.

JokerJohn committed:
remove unitest of issue1452

JokerJohn commented:
> Excellent, but could you then remove the (slow and redundant) unit test that basically does the same? Your example shows (a) it works, (b) a useful example.

done.

dellaert reviewed (approved):
Cool!
